ledger/participant-integration-api/src/main/scala/platform/partymanagement/PersistentPartyRecordStore.scala

```diff
@@ -161,6 +168,12 @@ class PersistentPartyRecordStore(
   private def doCreatePartyRecord(
       partyRecord: ParticipantParty.PartyRecord
   )(connection: Connection): Unit = {
+    if (
+      !ResourceAnnotationValidation
```

Comment:
Shouldn't it be calling `validateAnnotations`?

Reply:
Same comment applies throughout this PR.

Reply:
Here, at the store level, we need to check only the total size of annotations. More precisely, we only need this check when we update a party record (or a user). Nevertheless I added this check also at (a party record or party) creation time to have a consistent treatment of the annotation checks at the store level.

Full validation (i.e. length + syntax) will happen at the API service layer - that's my plan.